Use new v2 API to migrate v1 token to v2 token

blach opened the pull request.

blach committed:
Use new v2 API to migrate v1 token to v2 token

blach committed:
Use api.dropboxapi.com instead of api.dropbox.com for token migration call

timonus commented on TJDropbox/TJDropbox.m:
```diff
@@ -354,6 +339,33 @@ + (NSString *)asciiEncodeString:(NSString *const)string
+    // https://github.com/dropbox/dropbox-sdk-obj-c/blob/3769fbacb0b458de7c20db08edba6ca21c54b650/Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBSDKKeychain.m#L363-L453
+    // https://www.dropbox.com/developers/documentation/http/documentation#auth-token-from_oauth1
+    
+    NSMutableURLRequest *const request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"https://api.dropboxapi.com/2/auth/token/from_oauth1"]];
```
You could use `+apiRequestWithPath:accessToken:parameters:` to create the request complete with the host, `Content-Type`, and body you provided here. You'll need to modify `accessToken` to be optional, but otherwise would work.

blach replied:
Ok, will do.

blach committed:
Use +apiRequestWithPath:accessToken:parameters: in +migrateV1TokenToV2Token:accessTokenSecret:appKey:appSecret:completion:
Make accessToken optional in +apiRequestWithPath:accessToken:parameters: and return NSMutableURLRequest instead of NSURLRequest

blach commented:
I made the changes you requested and successfully tested it in my app.

blach committed:
Add create folder method

blach commented:
I have also added a create folder method

timonus commented:
Thanks @blach!
